staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -302,19 +302,19 @@ func (e *eventBroadcasterImpl) StartStructuredLogging(verbosity klog.Level) func
 			}
 			klog.V(verbosity).InfoS("Event occurred", "object", klog.KRef(event.Regarding.Namespace, event.Regarding.Name), "kind", event.Regarding.Kind, "apiVersion", event.Regarding.APIVersion, "type", event.Type, "reason", event.Reason, "action", event.Action, "note", event.Note)
 		})
+	if err != nil {
+		return nil
```

Comment:
This is risky - if anything is calling this function, calling `nil` will result in panic.

Ideally, this function should also return an error.
However, temporarily, you can add a TODO and instead here:
- log an error
- return `func() {}`

Reply:
Follow-up is in https://github.com/kubernetes/kubernetes/pull/120729.